core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -342,13 +343,20 @@ import scala.util.control.NonFatal
 
   private def receivePoll(p: Poll[_, _]): Unit =
     if (p.target == this) {
-      if (commitRefreshDeadline.exists(_.isOverdue())) {
+      if (commitRefreshDeadlines.exists(_._2.isOverdue())) {
+        val overdueTps = commitRefreshDeadlines.filter(_._2.isOverdue()).keySet
```

Comment:
`exists` and `filter` look like a repetition.

Reply:
Yes, look like, but the `if`-statement is guarding the whole section that should only run if there are overdue deadlines.

Reply:
Yes, please.